controllers/k8ssandra/replication.go

```diff
@@ -50,15 +50,20 @@ func (r *K8ssandraClusterReconciler) checkSystemReplication(ctx context.Context,
 // updateReplicationOfSystemKeyspaces ensures that the replication for the system_auth,
 // system_traces, and system_distributed keyspaces is up to date. It ensures that there are
 // replicas for each DC and that there is a max of 3 replicas per DC.
-func (r *K8ssandraClusterReconciler) updateReplicationOfSystemKeyspaces(ctx context.Context, kc *api.K8ssandraCluster, dc *cassdcapi.CassandraDatacenter, remoteClient client.Client, logger logr.Logger) result.ReconcileResult {
-	managementApiFacade, err := r.ManagementApi.NewManagementApiFacade(ctx, dc, remoteClient, logger)
+func (r *K8ssandraClusterReconciler) updateReplicationOfSystemKeyspaces(
+	ctx context.Context,
+	dcs []*cassdcapi.CassandraDatacenter,
+	remoteClient client.Client,
+	logger logr.Logger,
+) result.ReconcileResult {
+	managementApiFacade, err := r.ManagementApi.NewManagementApiFacade(ctx, dcs[0], remoteClient, logger)
 	if err != nil {
 		logger.Error(err, "Failed to create ManagementApiFacade")
 		return result.Error(err)
 	}
 
 	keyspaces := []string{"system_traces", "system_distributed", "system_auth"}
-	replication := cassandra.ComputeReplication(3, kc.Spec.Cassandra.Datacenters...)
+	replication := cassandra.ComputeReplication(3, dcs...)
```

Comment:
So, this changes the semantics slightly for multi-dc setups.

Previously: the keyspaces would be created with the desired replication for all dcs, including those not yet created.

Now: the keyspaces are created with the desired replication for the first dc, then updated to include replication for the second dc, and so on.

@jsanda do you think this is acceptable?

Reply:
This won't work. I somewhat explain it in https://github.com/k8ssandra/k8ssandra-operator/pull/262#discussion_r784222098. In short we would need to run repair on the system_auth keyspace on the second dc. Without doing that, the second dc won't be able to properly initialize.

Reply:
Sounds good, I will fix that.

Reply:
It's fixed now. Out of curiosity, why do we need to call this methods for every dc, wouldn't it be enough to call it for the first dc only?

Reply:
Great questions. There are a couple reasons it is called for every DC. First, we need to handle scale up within the DC. If we start with a single node cluster, system_auth is configured with RF = 1. If we then scale up to 3 nodes, it needs to be updated to RF = 3. Repair is needed as well, but that'll have to come post-1.0. The second reason has to do with adding a DC to an existing cluster. The are more, related changes around this in my add-dc branch. If we add dc2 to an existing cluster, again the RF needs to be updated.